### PR TITLE
Rebuild apidsl'd headers in cmake.

### PR DESCRIPTION
**Pull request by iphydf**
- Moved apidsl headers next to their generated versions. In the future,
  perhaps all (or most) headers will be apidsl-generated, so the sources
  should stay together.
- Try to find apidsl/apigen binary and astyle binary and use it for the
  format test. Don't run the format test if these can't be found.

---
This change is NOT reviewable. Use Github reviews instead.

### CODE REVIEW

File: cmake/ApiDsl.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  apidsl
+  apidsl.native
+  apidsl.byte
+  ${CMAKE_SOURCE_DIR}/../apidsl/apigen.native)
+find_program(ASTYLE NAMES
+  astyle)
+
+function(apidsl)
+  if(APIDSL AND ASTYLE)
+    foreach(in_file ${ARGN})
```
Comment (nurupo) [inline cmake/ApiDsl.cmake]: `s/in_file/api_file/g`, since you renamed `<name>.in.h` to `<name>.api.h`?

Comment (iphydf) [inline cmake/ApiDsl.cmake]: Yes.

**Comment by iphydf** [comment]
@nurupo done.

File: cmake/ApiDsl.cmake (hunk 1 above)
Comment (nurupo) [inline cmake/ApiDsl.cmake]: Ok.

File: toxcore/tox.api.h [hunk 2]
```diff
@@ -37,8 +37,8 @@ extern "C" {
 
 /*****************************************************************************
  * `tox.h` SHOULD *NOT* BE EDITED MANUALLY – any changes should be made to   *
- * `tox.in.h`, located in `other/apidsl/`. For instructions on how to        *
- * generate `tox.h` from `tox.in.h` please refer to `other/apidsl/README.md` *
+ * `tox.api.h`, located in `toxcore/`. For instructions on how to            *
+ * generate `tox.h` from `tox.api.h` please refer to `docs/apidsl.md`        *
  *****************************************************************************/
```
Comment (nurupo) [inline toxcore/tox.api.h]: Ok, here is `tox.api.h` with this change. Where is `tox.h` that is generated off this `tox.api.h` with this change?

Comment (iphydf) [inline toxcore/tox.api.h]: Done. Force pushed because I think this is the last change and I don't want to re-run travis yet another time.

**Pull request closed without merging**